internal/provider/resource_billing.go

```diff
@@ -0,0 +1,126 @@
+package provider
+
+import (
+	"context"
+	"time"
+
+	"github.com/PacketFabric/terraform-provider-packetfabric/internal/packetfabric"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceBilling() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceBillingCreate,
+		ReadContext:   resourceBillingRead,
+		UpdateContext: resourceBillingUpdate,
+		DeleteContext: resourceBillingDelete,
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Read:   schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"circuit_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				Description:  "Circuit ID of the order to modify.",
+			},
+			"subscription_term": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ValidateFunc: validation.IntInSlice([]int{1, 12, 24, 36}),
+				Description:  "The subscription term in months. Options are: 1, 12, 24, 36.",
+			},
+			"speed": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				Description:  "New speed of the circuit.",
+			},
+			"billing_product_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				Description:  "Billing type for VCs. Options are longhaul_dedicated, longhaul_usage, longhaul_hourly, surcharge, metro_dedicated.",
```

Comment:
@pontovinte actually here probably better to do:

`ValidateFunc: validation.StringInSlice([]string{"longhaul_dedicated", "longhaul_usage", "surcharge", "metro_dedicated"}, true),`

And remove the options from the description, right?